packages/runtime-common/index-writer.ts

```diff
@@ -250,6 +255,56 @@ export class Batch {
     return parseInt(total);
   }
 
+  private async updateRealmMeta() {
+    let results = await this.query([
+      `SELECT count(i.url) as total, i.display_names->>0 as display_name, i.types->>0 as code_ref
+       FROM boxel_index as i
+       INNER JOIN realm_versions r ON i.realm_url = r.realm_url
+          WHERE`,
+      ...every([
+        ['i.realm_url =', param(this.realmURL.href)],
+        ['i.type = ', param('instance')],
+        ['i.types IS NOT NULL'],
+        realmVersionExpression({ useWorkInProgressIndex: true }),
+      ]),
+      `GROUP BY i.display_names->>0, i.types->>0`,
+    ] as Expression);
+
+    let { nameExpressions, valueExpressions } = asExpressions(
+      {
+        key: RealmMetaKey.CardTypeSummary,
+        realm_url: this.realmURL.href,
+        realm_version: this.realmVersion,
+        value: results,
+        indexed_at: Date.now(),
+      } as Omit<RealmMetaTable, 'indexed_at'> & {
+        indexed_at: number;
+      },
+      {
+        jsonFields: ['value'],
+      },
+    );
+
+    await this.query([
+      ...upsert(
+        'realm_meta',
+        'realm_meta_pkey',
+        nameExpressions,
+        valueExpressions,
+      ),
+    ]);
+
+    await this.query([
+      `DELETE FROM realm_meta`,
+      'WHERE',
+      ...every([
+        ['key =', param(RealmMetaKey.CardTypeSummary)],
+        ['realm_version <', param(this.realmVersion)],
```

Comment:
this makes me question if we even need a realm version column. i guess for debugging purposes its probably useful information to have.